oidcserver/connector_test.go

```diff
@@ -31,12 +31,21 @@ type mockConnector struct {
 	// The returned identity.
 	Identity Identity
 
+	// preAuthenticate will be called if non-nil on each LoginPage with the
+	// passed loginrequest, and will return the identity it returns rather than
+	// the static value.
+	preAuthenticate func(lr LoginRequest) Identity
```

Comment:
Is it worth changing `mockConnector` to always use a function to get the Identity returned? We could implement the current functionality of returning a static identity with a function in all cases, I think.

Something like:

```go
type mockConnector struct {
  authFunc func(lr LoginRequest) Identity
}
```

For returning a static identity, something like `func(LoginRequest) { return staticIdentity }` is passed. The static `Identity` field is removed.

I don't feel super strongly about this, but I'm thinking it might reduce a little bit of cognitive load about how `mockConnector` works if there isn't conditional logic in `LoginPage`.